sycl/include/CL/sycl/sycl_span.hpp

```diff
@@ -608,7 +608,8 @@ as_writable_bytes(span<_Tp, _Extent> __s) noexcept
 }
 
 //  Deduction guides
-template <class _Tp, size_t _Sz> span(_Tp (&)[_Sz])->span<_Tp, _Sz>;
```

Comment:
I am not sure I understand the rationale for this change. Seems to me like the old version follows [`std::span`'s deduction guides](https://en.cppreference.com/w/cpp/container/span/deduction_guides).

Reply:
```int arr[] {1,2,3,4};
sycl::span fromArr {arr} ```  is not compiling correctly presently. And that's what we are trying to fix. 

The span<> specialization with a variable Extent does not have a constructor that accepts an array argument.  But the specialization that is tied to the constexpr dynamic_extent does. So this deduction guide correctly selects the correct specialization.  In that specialization, the actual Sz of the array is correctly used, overriding the dynamic_extent template arg. 

Given that there is only one array constructor and the deduction guide is only for array parameters, I think this should be a fairly safe change.

Reply:
I see. Would it make sense to add a comment about this? It seems a bit confusing to set `dymanic_extent` on something that isn't dynamic.

Reply:
~but my code is perfect as written~

Excellent idea. Done.